### DISCUSSION TITLE
Got error `Not an editor command: MasonUpdate` for first installation

### DISCUSSION THREAD

**Discussion by bavo96** [mason-org/mason.nvim#1183] [Q&A] [answered]
I use packer to install Mason via this command: 
```
  use {
    "williamboman/mason.nvim",
    run = ":MasonUpdate", -- :MasonUpdate updates registry contents
  }
```
For the first installation, I got an error: `Not an editor command: MasonUpdate`. After restarting neovim, everything's back to normal (because of Mason is successfully installed). How can I avoid this issue ? I think there's a related issue in nvim-treesitter about [TSUpdate](https://github.com/nvim-treesitter/nvim-treesitter/issues/3135) and they have fixed that. Can we have a similar solution for mason?

Many thanks.

**Comment by avegancafe** [comment, marked as answer]
Based on their fix, you should be able to run `require('mason.api.command').MasonUpdate()` @bavo96

**Reply by bavo96** [reply]
After using your suggestion, the error has gone and nothing is updated for the first installation. But for the next time I open nvim, I have to run `:MasonUpdate` again and there're some registries that are updating. Not a problem for me now but don't know if this is a correct answer, but i'll follow it by now :D

**Comment by zephod77** [comment]
I'm also getting the same error but my setup is different. This is not a fresh install and this used to work. I don't know when it broke.
In my ~/.config/nvim/init.vim file I have
```
call plug#begin('~/.config/nvim/autoload/plugged')
...
Plug 'williamboman/nvim-lsp-installer'
Plug 'williamboman/mason.nvim'
Plug 'williamboman/mason-lspconfig'
...
call plug#end()
...
lua << EOF
require("mason").setup()
require("nvim-autopairs").setup {}
EOF
```
Based on your answer above I added:
```
require('mason.api.command').MasonUpdate()
```
to the lua block above but then I get:
```
Error detected while processing /home/developer/.config/nvim/init.vim:
line 68:
E5108: Error executing lua [string ":lua"]:2: attempt to call field 'MasonUpdate' (a nil value)
stack traceback:
             [string ":lua"]:2: in main chunk
```
and :MasonUpdate still gives a E492: Not and editor command: MasonUpdate

**Reply by zephod77** [reply]
Interestingly,
```:Mason```
does work and if I type 'U', then all the installed packages were updated.
